openwisp_monitoring/monitoring/utils.py

```diff
@@ -3,60 +3,93 @@
-    # if database does not exists, create it
-    if settings.INFLUXDB_DATABASE not in databases:
-        db.create_database(settings.INFLUXDB_DATABASE)
-        logger.info(f'Created influxdb database {settings.INFLUXDB_DATABASE}')
+
+    def query(
+        self, query, params=None, epoch=None, expected_response_code=200, database=None
+    ):
+        """ Wrapper around ``InfluxDBClient.query()`` """
+        db = self.get_db()
+        database = database or TIMESERIES_DB['NAME']
+        return db.query(
+            query,
+            params,
+            epoch=epoch,
+            expected_response_code=expected_response_code,
+            database=database,
+        )
+
+    def write(
+        self,
+        name,
+        values,
+        tags=None,
+        timestamp=None,
+        database=None,
+        retention_policy=None,
+    ):
+        """ Method to be called via threading module. """
+        point = {'measurement': name, 'tags': tags, 'fields': values}
+        if isinstance(timestamp, datetime):
+            timestamp = timestamp.strftime('%Y-%m-%dT%H:%M:%SZ')
+        if timestamp:
+            point['time'] = timestamp
+        try:
+            self.get_db().write(
+                {'points': [point]},
+                {'db': database or TIMESERIES_DB['NAME'], 'rp': retention_policy},
+            )
+        except Exception as e:
+            raise e
+
+    def read(self, since, limit, order, extra_fields, key, fields, tags):
+        """ Method to be called via threading module. """
+        if extra_fields and extra_fields != '*':
+            fields = ', '.join([fields] + extra_fields)
+        elif extra_fields == '*':
+            fields = '*'
+        q = 'SELECT {fields} FROM {key}'.format(fields=fields, key=key)
+        conditions = []
+        if since:
+            conditions.append("time >= {0}".format(since))
+        if tags:
+            conditions.append(
+                ' AND '.join(["{0} = '{1}'".format(*tag) for tag in tags.items()])
+            )
+        if conditions:
+            conditions = 'WHERE %s' % ' AND '.join(conditions)
+            q = '{0} {1}'.format(q, conditions)
+        if order:
+            q = '{0} ORDER BY {1}'.format(q, order)
+        if limit:
+            q = '{0} LIMIT {1}'.format(q, limit)
+        return list(self.query(q, epoch='s').get_points())
+
+    def create_database(self):
+        """ creates database if necessary """
+        db = self.get_db()
+        response = db.query('SHOW DATABASES')
+        items = list(response.get_points('databases'))
+        databases = [database['name'] for database in items]
+        # if database does not exists, create it
+        if TIMESERIES_DB['NAME'] not in databases:
+            db.create_database(TIMESERIES_DB['NAME'])
+            logger.info(f'Created influxdb database {TIMESERIES_DB["NAME"]}')
+
+
+if TIMESERIES_DB['DATABASE'] == 'influxdb':
+    TimeseriesDB = InfluxDB()
```

Comment:
I think it's better if we have a generic TimeseriesDB class with a `db_type` attribute. This way you don't need to check `TIMESERIES_DB['DATABASE']` every time and you can just call `TimeseriesDB.method()`, which will check the database type and send the request